front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +59,44 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points as NonNullable<
+          PathProperties['operational_points']
+        >;
+        // Check if there are vias added by map click and insert them in the operational points
+        if (trainScheduleUsedForProjection.path.some((step) => 'track' in step)) {
+          trainScheduleUsedForProjection.path.forEach((step, i) => {
+            if ('track' in step) {
+              const positionOnPath = pathfindingResult.path_item_positions[i];
+              const indexToInsert = operationalPointsWithAllWaypoints.findIndex(
+                (op) => op.position >= positionOnPath
+              );
```

Comment:
What if the point on the map is the last one? We wouldn't find any OP whose position is greater than `positionOnPath` right? We should insert at the end of the list in that case I think?

Reply:
Hmm good point. And if I create a train with an OD added by map click, I get a white screen. There is an issue in the space time chart. Do you have it too ?

Reply:
Done, it also fixes the crash.